File: HW7/151044058_CSE241_HW7/PolygonVect.java
import java.awt.Color;
import java.awt.Graphics;
import java.util.ArrayList;

/**
 * 
 * @author Ali Haydar Kurban
 *
 */
public class PolygonVect extends Polygon 
{
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	private ArrayList<Point2D> polygonVect_Array;
	private double alan;
	private double cevre;
	public PolygonVect()
	{
		//Empty
	}
	
	/**
	 * 
	 * @param rec objesini PolygonVect ye donusturme constructorudur
	 */
	public PolygonVect(Rectangle rec)
	{
		polygonVect_Array = new ArrayList<Point2D>();
		
		Point2D p1 = new Point2D(rec.getCor_x(),rec.getCor_y());
		polygonVect_Array.add(p1);
		
		Point2D p2 = new Point2D(rec.getCor_x() + rec.getWidthh() ,rec.getCor_y());
		polygonVect_Array.add(p2);
		
		Point2D p3 = new Point2D(rec.getCor_x() + rec.getWidthh(),rec.getCor_y() + rec.getHeightt());
		polygonVect_Array.add(p3);
		
		Point2D p4 = new Point2D(rec.getCor_x(),rec.getCor_y() + rec.getHeightt());
		polygonVect_Array.add(p4);
		
		alan = rec.area();
		cevre = rec.perimeter();
	}
		
	/**
	 * 
	 * @param tri objesini PolygonVect ye donusturme constructorudur
	 */
	public PolygonVect(Triangle tri)
	{
		polygonVect_Array = new ArrayList<Point2D>();
		
		Point2D p1 = new Point2D(tri.getCor_x1(),tri.getCor_y1());
		polygonVect_Array.add(p1);
		
		Point2D p2 = new Point2D(tri.getCor_x2(),tri.getCor_y2());
		polygonVect_Array.add(p2);
		
		Point2D p3 = new Point2D(tri.getCor_x3(),tri.getCor_y3());
		polygonVect_Array.add(p3);
		
		alan = tri.area();
		cevre = tri.perimeter();
				
	}
	/**
	 * 
	 * @param cir objesini PolygonVect ye donusturme constructorudur
	 */
	public PolygonVect(Circle cir)
	{
		polygonVect_Array = new ArrayList<Point2D>();
		int i;
		int temp_size = 100;

		for(i = 1; i <= temp_size; ++i)
		{
			Point2D p1 = new Point2D(cir.getRadius()*2+ 2* (cir.getRadius()*Math.sin(((2.0*Math.PI)/100.0)*i)), cir.getRadius()*2+ 2*(cir.getRadius()*Math.cos(((2.0*Math.PI)/100.0)*i)));
			polygonVect_Array.add(p1);
		}
		
		alan = cir.area();
		cevre = cir.perimeter();
	}
	
	/**
	 * Alan fonksiyonu
	 */
	public double area()
	{
		return alan;
	}
	/**
	 * Cevre fonksiyonu
	 */
	public double perimeter()
	{
		return cevre;
	}
	

	@Override
	public void draw(Graphics g)
	{
		int [] array1 = new int [polygonVect_Array.size()];
		int [] array2 = new int [polygonVect_Array.size()];
		for(int i = 0; i < polygonVect_Array.size(); ++i)
		{
			array1[i] = (int)(polygonVect_Array.get(i)).getKoor_x();
			array2[i] = (int)(polygonVect_Array.get(i)).getKoor_y();
		}
		g.setColor(Color.BLUE);
		g.fillPolygon(array1, array2,polygonVect_Array.size());
	}

	@Override
	public void paintComponent(Graphics g)
	{
		super.paintComponent(g);
		draw(g);
	}

	/**
	 * Alanlarin kiyaslanmasina gore compareTo fonksiyonunun override hali
	 */
	@Override
	public int compareTo(Object arg0)
	{
		if(((Shape) arg0).area() > this.area())
			return -1;
		if(((Shape) arg0).area() < this.area())
			return 1;
		return 0;
	}
	/**
	 * 
	 */
	@Override
	public Shape increment()
	{
		return this;
	}
	/**
	 * 
	 */
	@Override
	public Shape decrement()
	{
		return null;
	}
}

File: HW5/151044058_CSE241_HW5/PolygonVect.cpp
#include "Circle.h"
#include "Triangle.h"
#include "Rectangle.h"
#include "Polygon.h"
#include "Shape.h"
#include "PolygonVect.h"

namespace HW5_151044058
{
	PolygonVect :: PolygonVect()
	{
		//Empty
	}
	
	PolygonVect :: PolygonVect(const Rectangle &rec) //	conversion rectangle
	{	
		Point2D p1(rec.getCor_x(),rec.getCor_y());
		vec_point2d.push_back(p1);
		Point2D p2(rec.getCor_x() + rec.getWidth() ,rec.getCor_y());
		vec_point2d.push_back(p2);
		Point2D p3(rec.getCor_x() + rec.getWidth(),rec.getCor_y() + rec.getHeight());
		vec_point2d.push_back(p3);
		Point2D p4(rec.getCor_x(),rec.getCor_y() + rec.getHeight());
		vec_point2d.push_back(p4);
		

	}

	PolygonVect :: PolygonVect(const Triangle &tri) //	conversion triangle
	{
		Point2D p1(tri.getCor_x1(),tri.getCor_y1());
		vec_point2d.push_back(p1);
		Point2D p2(tri.getCor_x2(),tri.getCor_y2());
		vec_point2d.push_back(p2);
		Point2D p3(tri.getCor_x3(),tri.getCor_y3());
		vec_point2d.push_back(p3);
	}

	PolygonVect :: PolygonVect(const Circle &cir)  //conversion circle
	{
		const double PI = M_PI;
		int i,temp_size = 100;

		for(i = 1; i <= temp_size; ++i)
		{
			Point2D p1(cir.getCenter_x()+(cir.getRadius()*sin(((2.0*PI)/100.0)*i)), cir.getCenter_y()+(cir.getRadius()*cos(((2.0*PI)/100.0)*i)));
			vec_point2d.push_back(p1);
		}
	}

	void PolygonVect :: draw_Shape(ostream& file)
	{
		int i;

		file << "<polygon points=\"";
		for(i = 0; i < this->vec_point2d.size(); ++i)
		{
			file << this->vec_point2d[i].getKoor_x()<<" "<< this->vec_point2d[i].getKoor_y()<<" ";	 
		}
		file<< "\" fill=\"black\" stroke=\"white\" />\n" ;

	}

	vector<PolygonVect :: Point2D> PolygonVect :: getVector_Point2D()
	{
		return vec_point2d;
	}
	
}